scipy/interpolate/fitpack.py

```diff
@@ -651,21 +413,15 @@ def splint(a, b, tck, full_output=0):
         on Numerical Analysis, Oxford University Press, 1993.
 
     """
-    t, c, k = tck
-    try:
-        c[0][0]
-        parametric = True
-    except:
-        parametric = False
-    if parametric:
-        return list(map(lambda c, a=a, b=b, t=t, k=k:
-                        splint(a, b, [t, c, k]), c))
+    if isinstance(tck, BSpline):
+        t, c, k = tck
+
+        # FITPACK expects the interpolation axis to be last, so roll it over
```

Comment:
This comment is a bit inaccurate --- FITPACK doesn't do the interpolation, instead it's done on Python side via `list(map(lambda c, t=t, k=k, mest=mest: sproot([t, c, k], mest), c))`, which is not really an optimal way to do it.

Again, I'd suggest deprecation warning for the multidim case, and a message suggesting `spl.integrate` instead.